src/include/btmem.h

```diff
@@ -412,8 +412,17 @@ struct __wt_page {
 	/*
 	 * Macros to copy/set the index because the name is obscured to ensure
 	 * the field isn't read multiple times.
+	 *
+	 * There are two versions of WT_INTL_INDEX_GET because the session split
+	 * generation is usually set, but it's not always required: for example,
+	 * if a page is locked for splitting, or being created or destroyed.
 	 */
-#define	WT_INTL_INDEX_COPY(page)	((page)->u.intl.__index)
+#define	WT_INTL_INDEX_GET_SAFE(page)					\
+	((page)->u.intl.__index)
+#define	WT_INTL_INDEX_GET(session, page, pindex) do {			\
```

Comment:
It bugs me that ```WT_INTL_INDEX_GET_SAFE``` and ```WT_INTL_INDEX_GET``` have different semantics. I see why, but I'd rather they both had ```pindex``` as a macro parameter.